### PR TITLE
update generic clients to support user-configurable runtime plugins

### PR DESCRIPTION
**Pull request by Velfi**
_This PR also updates `pre-commit ktlint` runner. Now it won't spit out a bazillion debug logs when run_

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/ClientRustSettings.kt [hunk 1]
```diff
@@ -156,6 +158,11 @@ data class ClientCodegenConfig(
                     ),
                     includeEndpointUrlConfig = node.get()
                         .getBooleanMemberOrDefault("includeEndpointUrlConfig", defaultIncludeEndpointUrlConfig),
+                    enableUserConfigurableRuntimePlugins =
+                    node.get().getBooleanMemberOrDefault(
```
Comment (jdisanti) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/ClientRustSettings.kt:162]: Nit: the indentation got weird here...

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ClientRuntimeTypesReExportGenerator.kt [hunk 2]
```diff
@@ -34,6 +34,10 @@ class ClientRuntimeTypesReExportGenerator(
                 "Interceptor" to RuntimeType.interceptor(rc),
                 "SharedInterceptor" to RuntimeType.sharedInterceptor(rc),
             )
+
+            if (codegenContext.enableUserConfigurableRuntimePlugins) {
+                rustTemplate("pub use #{runtime_plugin}::{RuntimePlugin, SharedRuntimePlugin};", "runtime_plugin" to RuntimeType.smithyRuntimeApi(rc).resolve("client::runtime_plugin"))
```
Comment (jdisanti) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ClientRuntimeTypesReExportGenerator.kt:39]: Good thinking!

**Review by ysaito1001 — COMMENTED** [review]
LGTM! Maybe we also want to add a Kotlin unit test (like a compile-only test) to see if runtime plugins are configurable?

**Review by jdisanti — APPROVED** [review]

**Comment by jdisanti** [comment]
Oh, yeah, a Kotlin unit test wouldn't be a bad idea.

**Review by rcoh — APPROVED** [review]

**Pull request merged**